### PR TITLE
creating a map including geojson data

### PR DESCRIPTION
**Pull request by MackyDIARRA**
I had to import the zones.json file which allows me to retrieve the coordinates of each country mentioned on the map but I cannot manage to create the interface which makes it possible to concretely open the plotted data corresponding to each country

### CODE REVIEW

File: Untitled.ipynb [hunk 1]
```diff
@@ -7,28 +7,98 @@
...
+    "\n",
     "\n",
     "parsers_submodules = [module.name for module in iter_modules(lib.__path__)]\n",
     "print(parsers_submodules)\n",
     "\n",
     "country_parser = importlib.import_module('lib.{}'.format(parsers_submodules[3]))\n",
-    "print(country_parser)"
+    "print(country_parser)\n",
+    "\n",
+    "find = str(input(\"veuillez rentrer le code du pays correspondant : \"))\n",
```
Comment (martinRenou) [inline Untitled.ipynb:47]: For now you're using `input`, but you should probably remove that and to this programmatically when clicking on the map

File: Untitled.ipynb [hunk 2]
```diff
@@ -7,28 +7,98 @@
...
+    "time_data = np.array(param2,dtype='datetime64')\n",
+    "\n",
+    "\n",
+    "fig = plt.figure(title=\"Electricity production\")\n",
+    "\n",
+    "axes_options = {\"x\": {\"label\": \"time \"}, \"y\": {\"label\": \"production (MWh) \",\"label_offset\" : \"6ex\"}}\n",
+    "bar1 =  plt.bar(time_data,production_data,colors=CATEGORY10, labels=[\"oil\", \"Biomass\", \"Coal\",\"Wind\",\"Hydro\",\"Gas\",\"\",\"Nuclear\"],display_legend=True,)\n",
+    "axes = plt.axes(options = axes_options)\n",
+    "\n",
+    "fig\n",
```
Comment (martinRenou) [inline Untitled.ipynb:71]: Suggestion: Make this part a function that you can reuse later

File: Untitled.ipynb [hunk 3]
```diff
@@ -7,28 +7,98 @@
...
+    "        f.write(r.content.decode(\"utf-8\"))\n",
+    "\n",
+    "with open(\"./eu-countries.geo.json\") as f:\n",
+    "    data = json.load(f)\n",
+    "\n",
+    "m = Map(center=(50.6252978589571, 0.34580993652344), zoom=3)\n",
+    "\n",
+    "geo_json = GeoJSON(data=data,style={'opacity': 1, 'dashArray': '9', 'fillOpacity': 0.1, 'weight': 1 },hover_style={'color': 'white', 'dashArray': '0', 'fillOpacity': 0.5},style_callback=random_color)\n",
+    "m.add_layer(geo_json)\n",
+    "\n",
```
Comment (martinRenou) [inline Untitled.ipynb:92]: I guess you're missing the part that creates the plot when clicking on the map now :)

I'd suggest to look at using this method: https://github.com/jupyter-widgets/ipyleaflet/blob/master/ipyleaflet/leaflet.py#L1243

Comment (martinRenou) [reply Untitled.ipynb:92]: There is a similar dashboard to what you are trying to achieve here: https://github.com/jupyter-widgets/ipyleaflet/blob/master/examples/WealthOfNations.ipynb

The only difference is that they are using `on_hover` and not `on_click`

File: Trees.ipynb [hunk 4]
```diff
@@ -0,0 +1,265 @@
+{
```
Comment (martinRenou) [inline Trees.ipynb:1]: This might have been committed by mistake?

**Comment by MackyDIARRA** [comment]
I have been renamed my "untitled.ipynb" notebook to "Electricity_map.ipynb". I tried to adapt https://github.com/jupyter-widgets/ipyleaflet/blob/master/examples/WealthOfNations.ipynb notebook in my notebook but I can't find a solution that allows me to display the plot of each country when I click on a country on the map

File: Trees.ipynb (hunk 4 above)
Comment (MackyDIARRA) [reply Trees.ipynb:1]: no I've submitted my progress

Comment (MackyDIARRA) [reply Trees.ipynb:1]: aah yes for this notebook

File: Electricity_map.ipynb [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    "geo = GeoJSON(data=countries,style={\"fillColor\": \"white\", \"weight\": 0.5},hover_style={\"fillColor\": \"#1f77b4\"},name=\"Countries\")\n",
+    "m.add_layer(geo)\n",
+    "\n",
+    "\n",
+    "widget_control1 = WidgetControl(widget=b, position=\"bottomright\")\n",
+    "m.add_control(widget_control1)\n",
+    "\n",
+    "\n",
+    "def on_click(event, feature, **kwargs):\n",
+    "    if country == \"click\" :\n",
```
Comment (martinRenou) [inline Electricity_map.ipynb:95]: The country variable is not defined?

File: Electricity_map.ipynb [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    "import requests\n",
+    "from ipyleaflet import Map, GeoJSON, WidgetControl\n",
+    "import ipywidgets as widgets\n",
+    "\n",
+    "\n",
+    "\n",
+    "find = str(input(\"veuillez rentrer le code du pays correspondant : \"))\n",
+    "a = fetch_production(find)\n",
+    "     \n",
+    "def mapping(param1) :\n",
```
Comment (martinRenou) [inline Electricity_map.ipynb:54]: Your function creates a bar chart, "mapping" does not seem like a proper name for it?

**Review by martinRenou — CHANGES_REQUESTED** [review]

File: Electricity_map.ipynb (hunk 5 above)
Comment (MackyDIARRA) [reply Electricity_map.ipynb:95]: sorry I wanted to put type not "country"

File: Electricity_map.ipynb (hunk 6 above)
Comment (MackyDIARRA) [reply Electricity_map.ipynb:54]: It was supposed to be a test but I will modify it

File: Electricity_map.ipynb (hunk 5 above)
Comment (martinRenou) [reply Electricity_map.ipynb:95]: `type` is also not defined, that might not be what you want to do. At this point in the code you already know that it's a click event, so this condition might not be needed :)

Comment (MackyDIARRA) [reply Electricity_map.ipynb:95]: okay I just got it

**Review by MackyDIARRA — COMMENTED** [review]
I deleted every unnecessary files which I don't need

File: Electricity_map.ipynb [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    "            param.append([subdata['production'][production] for subdata in data])\n",
+    "            labels.append(production)\n",
+    "    \n",
+    "    \n",
+    "    update_figure(param, labels , time_data ,feature['properties']['name'])\n",
+    "\n",
+    "        \n",
+    "geo.on_click(update_geojson)\n",
+    "\"\"\"\n",
+    "Element = {'Map': app , 'Figure': fig}\n",
```
Comment (martinRenou) [inline Electricity_map.ipynb:137]: The 'Map' pane you are defining here uses the `app` variable which contains the bqplot figure `fig`, so you are using the figure twice in the flex layout?

**Review by MackyDIARRA — COMMENTED** [review]
I created a new notebook that can contribute for the making of my dashboard in my notebook Electricity_map.ipynb

File: Electricity_map.ipynb [hunk 8]
```diff
@@ -88,14 +74,14 @@
     "    fig.title = country\n",
     "    \n",
     "\n",
-    "    \n",
-    "\n",
     "\n",
     "def on_click(event, feature, **kwargs):\n",
     "  \n",
     "    data = fetch_production(feature['properties']['iso_a2'])\n",
+    "    data2 = fetch_price()\n",
```
Comment (martinRenou) [inline Electricity_map.ipynb:81]: This makes the callback fail on my side. You are not using the `data2` variable, so I would remove this line for now.

Comment (MackyDIARRA) [reply Electricity_map.ipynb:81]: sorry I will delete it too

**Review by MackyDIARRA — COMMENTED** [review]
I deleted a  data2 and price data variable

**Comment by martinRenou** [comment]
There is another occurrence of `data2` in your Notebook that you might have missed? The one pointed by my comment above

**Pull request merged**